docs/src/main/paradox/cdi.md

```diff
@@ -0,0 +1,72 @@
+# Context Injection
+
+As we saw in the Getting Started example, it is perfectly possible to inject any Bean available in Spring into a 
+Cloudstate entity class.
+It is also possible to use the annotations present in the javax.inject package 
+([JSR330](https://jcp.org/en/jsr/detail?id=330)). 
+The Cloudstate Springboot Support library already includes the necessary dependencies so you don't have to worry about it.
+
+You can annotate your entity classes with Spring @Component or @Service annotations but we have created a convenient 
+annotation that we call @CloudstateEntityBean that can be used for that too.
+
+## JSR330
+
+The Cloudastate Springboot Support library supports JSR330 within the scope of the support provided by Spring itself to 
+this specification.
+Note that the Cloudstate Java Support library on which we depend allows you to bind Cloudstate and any other DI container 
+you want. However, no specific module for any of these other containers has yet been made.
+
+Feel free to contribute or suggest support for more runtimes.
+
+## Injecting EntityId and Cloudstate Context Objects
+
+You can use the @EntityId annotation to access the managed entity's id.
+It is also possible to have access to the EventSourcedEntityCreationContext created during the activation of the object 
+by Cloudstate. However for this you will need to annotate the Context property with the annotation @CloudstateContext 
+as in the example below:
+
+```java
+@EntityId
+private String entityId;
+
+@CloudstateContext
+private EventSourcedContext context;
+```
+
+### Using properties instead constructors
+
+It is currently not possible to inject Cloudstate's EntityId and Context properties via constructor. 
+This is because the life cycles of these priorities differ from the life cycle of objects managed directly by Spring.
+
+This is obviously only an issue if you want to inject EntityId or EventSourcedEntityCreationContext.
+Otherwise, if you want to inject only other Spring Context Beans, you can use injection via builders as normal.
+
+The builders below would be perfectly acceptable:
+
+```java
+@EventSourcedEntity
+@CloudstateEntityBean
+public final class ShoppingCartEntity {
+    private final Map<String, Shoppingcart.LineItem> cart = new LinkedHashMap<>();
+
+    @EntityId
+    private String entityId;
+
+    @CloudstateContext
+    private EventSourcedContext context;
+
+    private final RuleService ruleService;
```

Comment:
like the constructor injection here; different on other occasions (see my other comments)

Reply:
In this case `@CloudstateContext` is used to assist the discovery engine of the springboot-support and not Spring itself. This is necessary because this type of data cannot be managed through the life cycle of Spring objects. I think I mention this in the documentation in the Conventions and Restrictions section. It could not be declared without this annotation or otherwise as it would probably cause an NPE at some point in the execution when that property was referenced